src/FakeItEasy/Core/TypeCatalogue.cs

```diff
@@ -91,7 +91,7 @@ private static IEnumerable<Assembly> GetAllAssemblies(IEnumerable<string> extraA
                     // Exclude the ReflectionOnly assemblies because we may fully load them later.
                     .Where(a => !a.ReflectionOnly)
 #endif
-                    .Where(a => !a.IsDynamic)
+                    .Where(a => !a.IsDynamic && !a.Name().Equals("DynamicProxyGenAssembly2", StringComparison.OrdinalIgnoreCase))

```

Comment:
> Have you seen the error in the wild, or only while running the integration tests?

Only while running the integration tests, and only about 1 time out of 5 or 10

> Do you know where the error was triggered? Is it the same as in #1092, where you see problems loading a particular type?

I don't know where exactly. It is what prompted #1092, but it's not the same.

> I think this is happening when DynamicProxyGenAssembly2 references FakeItEasy

Doesn't it always reference FakeItEasy? As soon as we create a fake, at least?

> What think you?

This might be the explanation. I'll have a look later today.